test/functional/page_objects/dashboard_page.js

```diff
@@ -550,20 +551,10 @@ export function DashboardPageProvider({ getService, getPageObjects }) {
 
     async waitForRenderComplete() {
       await retry.try(async () => {
-        const sharedItems = await find.allByCssSelector('[data-shared-item]');
-        const renderComplete = await Promise.all(sharedItems.map(async sharedItem => {
-          return await sharedItem.getAttribute('data-render-complete');
+        const sharedItems = await this.getPanelSharedItemData();
+        await Promise.all(sharedItems.map(async sharedItem => {
+          return await visualization.waitForRender(sharedItem.element, sharedItem.title, { ignoreNonVisualization: true });
```

Comment:
I think this might be a lot slower than the previous implementation.  The element being passed in here, I don't *think* is the element with the `TEST_SUBJECT_VISUALIZE` on it, so for every panel on the dashboard, it's going to call `testSubjects.findAllDescendant(TEST_SUBJECT_VISUALIZE, parentElement);`.  All the `findAll*` functions are really slow and should be avoided where possible.

I could be wrong though...

Could you maybe create a `waitForAllToRender` function that instead of looking for a single visualization, just always does:

```
const visualizations = await testSubjects.findAllDescendant(TEST_SUBJECT_VISUALIZE, bodyElement);
```

then loops through them all and waits for `'data-loading'` attribute to go away?

Reply:
Actually it IS that element (why I needed to change the function to also accept the element itself and not just look in its descendant) :-) Would you still want it to changed in that case?